db/blip_handler_collections.go

```diff
@@ -0,0 +1,94 @@
+package db
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/couchbase/go-blip"
+	"github.com/couchbase/sync_gateway/base"
+)
+
+func isDefaultScopeAndCollection(scope, collection string) bool {
+	return scope == base.DefaultScope && collection == base.DefaultCollection
+}
+
+func parseScopeAndCollection(sc string) (scope, collection *string, err error) {
+
+	parts := strings.Split(sc, base.ScopeCollectionSeparator)
+	switch len(parts) {
+	case 1:
+		scope = base.StringPtr(base.DefaultScope)
+		collection = &parts[0]
+	case 2:
+		scope = &parts[0]
+		collection = &parts[1]
+	default:
+		return nil, nil, fmt.Errorf("unknown keyspace format: %q - expected 1-2 fields", sc)
+	}
+
+	if *scope == "" {
+		return nil, nil, fmt.Errorf("Scope in [scope.]collection can not be empty %q", sc)
+	}
+
+	if *collection == "" {
+		return nil, nil, fmt.Errorf("Collection in [scope.]collection can not be empty %q", sc)
+	}
+
+	return scope, collection, nil
+}
+
+func (bh *blipHandler) handleGetCollections(rq *blip.Message) error {
+	var requestBody GetCollectionsRequestBody
+	err := rq.ReadJSONBody(&requestBody)
+	if err != nil {
+		return fmt.Errorf("Error deserializing json Sync err: %w body: %s", err, requestBody)
+	}
+	bh.logEndpointEntry(rq.Profile(), requestBody.String())
+
+	if len(requestBody.Collections) != len(requestBody.CheckpointIDs) {
+		// TODO: can I send the information back to the client, is this leaking data? Should I log?
```

Comment:
You can tell them what was wrong with their request (mismatched array lengths) without leaking data - I don't think there's much more that needs to be sent.